addons/account/models/account_move.py

```diff
@@ -5049,6 +5049,18 @@ def _disable_recursion(self, container, key, default=None, target=True):
     def _mailing_get_default_domain(self, mailing):
         return ['&', ('move_type', '=', 'out_invoice'), ('state', '=', 'posted')]
 
+    @api.model
+    def _routing_check_route(self, message, message_dict, route, raise_exception=True):
+        if ((route and route[2]) or {}).get('move_type', 'entry') in ('out_invoice', 'in_invoice') and len(message_dict['attachments']) < 1:
+            # Don't create the move if no attachment.
+            mail_template = self.env.ref('account.email_template_mail_gateway_failed')
+            mail_template.send_mail(self.env.company.id, force_send=True, email_values={
+                'email_to': message_dict['from'],
+                'email_from': tools.formataddr(('MAILER-DAEMON', self.env.company.bounce_email)),
+            })
```

Comment:
Might be better to call `_generate_template` on the template to get the body/subject and then call `_routing_create_bounce_email`, this way it can figure out exactly what email to use (bounce_email isn't required)

Reply:
It's done :+1: